extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/health/MongoHealthCheck.java

```diff
@@ -28,8 +37,12 @@
     public static final String CLIENT_DEFAULT = "<default>";
     public static final String CLIENT_DEFAULT_REACTIVE = "<default-reactive>";
 
-    private Map<String, MongoClient> clients = new HashMap<>();
-    private Map<String, ReactiveMongoClient> reactiveClients = new HashMap<>();
+    private final Map<String, MongoClient> clients = new HashMap<>();
+    private final Map<String, ReactiveMongoClient> reactiveClients = new HashMap<>();
+
+    @ConfigProperty(name = "quarkus.mongodb.health.database", defaultValue = "admin")
```

Comment:
This is an hidden property as you didn't define it inside the MongoClient configuration object.
I'd rather add this config item to the MongoClient configuration object and add the default value there, this will also allow to add a documentation.

Reply:
+1, that's just a quick hack to be sure we agree on that. 

I wanted to get a timeout too, but not sure which one to use? read timeout or heartbeat?

Reply:
read timeout seems to be right here.

But timeout on healthcheck is not straightforward as their could be multiple health checks (mongo, hibernate, vault) on the same application so maybe it should be done globally on health check implementation not check by check.
And if using some orchestration tool to call the health check, it can have a timeout at its side.

Reply:
There are multiple issues. Adding a reasonable timeout is a defensive approach to avoid thread starvation.

* You must not keep the thread block for more than what you expect to be a failure (right now it's waiting indefinitely and you may end up in a situation where it's really really long - high latency, corrupted data... yes the client should handle that... or not)
* While the infrastructure has a timeout, it's not the case everywhere - the dev UI has a different timeout (I need to check to be sure it has one actually)
* I believe SmallRye Health Check has a global timeout (@xstefank can you confirm?) - however, a blocked thread is a blocked thread and even if the check reports the failure, you may have a blocked thread.

Reply:
BTW, I pushed the config change

Reply:
Agree that adding a timeout at this level is needed to catch all possible issue.